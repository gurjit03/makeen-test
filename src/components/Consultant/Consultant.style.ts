import styled from 'styled-components';
import { fontSizes, verticalSpacing } from '../../config/jss';

export const CWrapper = styled.li`
    display: flex;
    align-items: center;
    list-style: none;
    margin-bottom: ${verticalSpacing * 2}px;
`;

export const CAvatar = styled.img`
    max-width: 60px;
    border-radius: 50%;
    display: inline-block;
    margin-right: ${verticalSpacing * 2}px;
`

export const CName = styled.p`
    font-size: ${fontSizes.large}px;
`;