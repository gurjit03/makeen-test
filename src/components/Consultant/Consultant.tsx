import React from 'react'
import { Consultant as IConsulant } from '../../interfaces/Consultant.interface'
import { CWrapper, CAvatar, CName } from './Consultant.style'

interface ConsultantProps extends IConsulant {
	wrapperComponent?: React.FC
}

const ConsultantWrapper: React.FC = ({ children }) => {
	return <CWrapper>{children}</CWrapper>
}

const Consultant: React.FC<ConsultantProps> = props => {
	const {
		wrapperComponent: Wrapper = ConsultantWrapper,
		avatar,
		firstName,
		lastName,
	} = props
	const name = `${firstName} ${lastName}`
	return (
		<Wrapper>
			<CAvatar src={avatar} />
			<CName>{name}</CName>
		</Wrapper>
	)
}

export default Consultant
