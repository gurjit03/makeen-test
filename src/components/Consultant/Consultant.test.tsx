import React from 'react';
import { render, screen } from '@testing-library/react'
import consultantsMock from '../../mock/consultant.mock';
import Consultant from './Consultant';

describe('<Consultant />', () => {
    it('should render properly', () => {
        const { asFragment } = render(<Consultant {...consultantsMock[1]} />)
        expect(asFragment()).toMatchSnapshot();
    });

    it('should have consultant data', () => {
        render(<Consultant {...consultantsMock[1]} />)
        const avatarName = `${consultantsMock[1].firstName} ${consultantsMock[1].lastName}`;
        const avatarNameEl = screen.getByText(avatarName)
        const avatarImgEl = screen.getByAltText('avatar');
        expect(avatarNameEl).toBeInTheDocument()
        expect(avatarImgEl).toBeInTheDocument();
        expect(avatarImgEl.src).toContain("https://avataaars.io/?avatarStyle=Circle&topType=LongHairStraight&accessoriesType=Blank&hairColor=BrownDark")
    })
})