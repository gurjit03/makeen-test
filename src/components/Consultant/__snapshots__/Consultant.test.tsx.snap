// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Consultant /> should render properly 1`] = `
<DocumentFragment>
  <li
    class="sc-bdfBQB dSprCg"
  >
    <img
      alt="avatar"
      class="sc-gsTEea dtNHVy"
      src="https://avataaars.io/?avatarStyle=Circle&topType=LongHairStraight&accessoriesType=Blank&hairColor=BrownDark&facialHairType=Blank&clotheType=BlazerShirt&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light'"
    />
    <p
      class="sc-dlfnuX iHPnGy"
    >
      john doe2
    </p>
  </li>
</DocumentFragment>
`;
